tests/simulator_test.cpp

```diff
@@ -64,48 +59,68 @@ class EnergyCollectorTest : public ::testing::Test {
     return false;
   }
 
-  void
-  printCollectors(const std::vector<std::unique_ptr<objects::EnergyCollector>>
-                      &energyCollectors) const {
+  float getCollectorRadius(const Collectors &energyCollectors) const {
+    return energyCollectors[0]->getRadius();
+  }
+
+  // returns value of angle between two neighborhood collectors and
+  // Vec3(0, 0, 0)
+  float getCollectorAngle(const Collectors &energyCollectors) const {
+    return 2 * kPi /
+           (energyCollectors.size() + (energyCollectors.size() % 2) - 2);
+  }
+
+  void printCollectors(const Collectors &energyCollectors) const {
+    int i = 0;
     for (const auto &collector : energyCollectors) {
-      std::cout << "x: " << collector->getOrigin().x()
-                << " y: " << collector->getOrigin().y()
-                << " z: " << collector->getOrigin().z()
-                << " rad: " << collector->getRadius() << "\n";
+      std::cout << i++ << " " << *collector << std::endl;
     }
   }
 
-  float topCollectorZCoordinate(
-      const std::vector<std::unique_ptr<objects::EnergyCollector>>
-          &energyCollectors) {
+  float getMaxZ(const Collectors &energyCollectors) {
     float maxZ = 0;
     for (const auto &collector : energyCollectors) {
-      if (collector->getOrigin().z() > maxZ) {
-        maxZ = collector->getOrigin().z();
-      }
+      maxZ = std::max(maxZ, collector->getOrigin().z());
     }
     return maxZ;
   }
+
+  float getCollectorPositionRadius(const Collectors &energyCollectors) {
+    float maxX = 0;
+    for (const auto &collector : energyCollectors) {
+      maxX = std::max(maxX, collector->getOrigin().x());
+    }
+    return maxX;
+  }
+
+  // |xRotation| must be in radians
```

Comment:
Ad.1 Because Im used to it so much. I've never seen trygonometric function in acoustics that accepts degrees not radians, tahts why I would like to keep this convention. 

Ad 2-4 I added static methond in ray that deos that in spherical coordinates ;)